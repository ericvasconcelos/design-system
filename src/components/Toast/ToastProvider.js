import React, { createContext, useContext, useEffect, useReducer } from 'react'

import { node } from 'prop-types'

const ToastContext = createContext()
export const useToast = () => useContext(ToastContext)

export const ToastContextProvider = ({ children }) => {
  const reducer = (state, nextState = { action: undefined, payload: undefined }) => {
    const { action, payload } = nextState
    switch (action) {
      case 'SET_TEXT':
        return { ...state, text: payload.text }
      case 'SET_ACTIVE':
        return { ...state, active: payload.active }
      case 'SET_SECONDS':
        return { ...state, seconds: payload.seconds }
      case 'SET_ONTIMEOUT':
        return { ...state, onTimeout: payload.onTimeout }
      case 'SET_STATE':
        return { ...state, ...payload }
      default:
        break
    }
  }
  const toastDefaults = {
    active: false,
    styleType: 'success',
    text: '',
    seconds: 4,
    onTimeout: () => {}
  }

  const [toast, dispatch] = useReducer(reducer, toastDefaults)
  const setText = text => {
    dispatch({ action: 'SET_TEXT', payload: { text } })
  }
  const setActive = active => {
    dispatch({ action: 'SET_ACTIVE', payload: { active } })
  }
  const setOnTimeout = onTimeout => {
    dispatch({ action: 'SET_ONTIMEOUT', payload: { onTimeout } })
  }
  const setToastState = (state = {}) => {
    dispatch({
      action: 'SET_STATE',
      payload: { ...toastDefaults, onTimeout: () => setActive(false), ...state }
    })
  }

  useEffect(() => {
    setOnTimeout(() => setActive(false))
  }, [])


  return (
    <ToastContext.Provider value={{ toast, setToastState, setText, setActive, setOnTimeout }}>
      {children}
    </ToastContext.Provider>
  )
}

ToastContextProvider.propTypes = {
  children: node.isRequired
}
