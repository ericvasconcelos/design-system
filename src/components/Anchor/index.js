import React from 'react'

import dynamic from 'next/dynamic'
import NextLink from 'next/link'
import { bool, element, elementType, node, oneOf, oneOfType, string, func } from 'prop-types'
import { buttonStyles } from 'utils/getButtonStyle/data'
import { getIconNames } from 'utils/getIconNames'

import { Wrapper } from './styles'

const Icon = dynamic(() => import('../Icon/index.js').then(module => module.Icon), {
  ssr: false
})

export const Anchor = ({
  children,
  styleColor,
  href,
  icon,
  id,
  styleType,
  size,
  borderRadius,
  passHref,
  download,
  expanded,
  className,
  target,
  rel,
  title,
  onClick
}) => (
  <NextLink href={href} passHref={passHref} className={className}>
    <Wrapper
      data-testid="Link"
      styleColor={styleColor}
      className={className}
      {...(id && { id })}
      styleType={styleType}
      size={size}
      borderRadius={borderRadius}
      download={download}
      expanded={expanded}
      {...(target && { target })}
      {...(rel && { rel })}
      {...(title && { title })}
      {...(onClick && { onClick })}
    >
      {children}
      {icon && Icon && (
        <Icon icon={icon} styleColor={buttonStyles[styleType][styleColor].fontColor} />
      )}
    </Wrapper>
  </NextLink>
)

Anchor.propTypes = {
  children: oneOfType([node, element, elementType, string]),
  styleColor: oneOf(['blue', 'white', 'dark']),
  icon: oneOf([...getIconNames(), '']),
  id: string,
  href: string,
  styleType: oneOf(['primary', 'secondary', 'link', 'text']),
  size: oneOf(['small', 'medium', 'large']),
  borderRadius: oneOf(['default', 'rounded']),
  passHref: bool,
  download: bool,
  expanded: bool,
  className: string,
  target: string,
  rel: string,
  title: string,
  onClick: func
}

Anchor.defaultProps = {
  styleColor: 'blue',
  icon: '',
  href: '',
  styleType: 'link',
  size: 'medium',
  borderRadius: 'default',
  passHref: false,
  id: '',
  download: false,
  expanded: false,
  className: '',
  children: '',
  target: '',
  rel: '',
  title: '',
  onClick: null
}
