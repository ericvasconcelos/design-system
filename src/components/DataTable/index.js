import React, { useRef, useState } from 'react'

import { Icon } from 'components/Icon'
import { string, arrayOf, shape, bool, func, number, object } from 'prop-types'

import { RowTable } from './RowTable'
import {
  WrapperTable,
  Table,
  Header,
  HeaderContent,
  IconSort,
  Pagination,
  ShowMoreButton,
  Loader,
  Text
} from './styles'

/** Componente de tabela é utilizado para exibir dados em forma de tabela */

export const DataTable = ({
  headers,
  collapse,
  data,
  numberPages,
  currentPage,
  onChangePage,
  onSort,
  showMoreItemsText,
  isLoading,
  isPageLoading,
  messageWhenEmpty,
  className
}) => {
  const tbodyRef = useRef(null)
  const theadRef = useRef(null)
  const cellRefs = useRef([])
  const [currentRowCell, setCurrentRowCell] = useState([0, 0])

  const handleFocus = (currentRow, currentCell) => {
    setCurrentRowCell([currentRow, currentCell])
  }

  const tableKeyDownHandler = (event, indexRow, indexCel) => {
    const rowsLength = tbodyRef.current.children.length
    const cellLength = theadRef.current.children.length

    const handleArrowRight = () => {
      event.preventDefault()
      if (indexCel < cellLength - 1) cellRefs.current[indexRow][indexCel + 1].focus()
    }
    const handleArrowLeft = () => {
      event.preventDefault()
      if (indexCel > 0) cellRefs.current[indexRow][indexCel - 1].focus()
    }
    const handleArrowDown = () => {
      event.preventDefault()
      if (indexRow < rowsLength - 1) cellRefs.current[indexRow + 1][indexCel].focus()
    }
    const handleArrowUp = () => {
      event.preventDefault()
      if (indexRow > 0) cellRefs.current[indexRow - 1][indexCel].focus()
    }
    const handleHome = () => {
      event.preventDefault()
      if (!event.ctrlKey) {
        return cellRefs.current[indexRow][0].focus()
      }
      cellRefs.current[0][0].focus()
    }

    const handleEnd = () => {
      event.preventDefault()
      if (!event.ctrlKey) {
        return cellRefs.current[indexRow][cellLength - 1].focus()
      }
      cellRefs.current[rowsLength - 1][cellLength - 1].focus()
    }

    const keyToAction = {
      ArrowRight: handleArrowRight,
      ArrowLeft: handleArrowLeft,
      ArrowDown: handleArrowDown,
      ArrowUp: handleArrowUp,
      Home: handleHome,
      End: handleEnd,
      Default: () => {}
    }

    return (keyToAction[event.key] || keyToAction.Default)()
  }

  return (
    <>
      <WrapperTable data-testid="DataTable" className={className}>
        <Table role="grid" aria-rowcount={data.length}>
          <thead>
            <tr ref={theadRef} role="row">
              {headers.map(header => (
                <Header
                  role="columnheader"
                  onClick={header.sortable ? () => onSort(header.key) : null}
                  sortable={header.sortable}
                  key={header.key}
                  id={`header-${header.key}`}
                >
                  <HeaderContent>
                    {header.label}
                    {header.sortable && (
                      <IconSort icon="chevronDown" styleColor="darkBlueCash500" />
                    )}
                  </HeaderContent>
                </Header>
              ))}
              {collapse.length !== 0 && <Header key="collapse" isCollapse />}
            </tr>
          </thead>
          {!isLoading && (
            <tbody ref={tbodyRef}>
              {data.map((row, index) => (
                <RowTable
                  key={row.key}
                  index={index}
                  data={row}
                  headers={headers}
                  collapse={collapse}
                  currentRowCell={currentRowCell}
                  handleFocus={handleFocus}
                  cellRefs={cellRefs}
                  tableKeyDownHandler={tableKeyDownHandler}
                />
              ))}
            </tbody>
          )}
        </Table>
        {data.length === 0 && !isLoading && messageWhenEmpty && <Text>{messageWhenEmpty}</Text>}
        {isLoading && <Loader />}
      </WrapperTable>
      {numberPages > 1 && currentPage < numberPages && (
        <Pagination>
          <ShowMoreButton disabled={isPageLoading} onClick={() => onChangePage(currentPage + 1)}>
            {isPageLoading ? (
              <Icon icon="loader" styleColor="lightBlueCash300" />
            ) : (
              `${showMoreItemsText}`
            )}
          </ShowMoreButton>
        </Pagination>
      )}
    </>
  )
}

DataTable.propTypes = {
  /** conteúdo da cabeçalho */
  headers: arrayOf(
    shape({
      key: string.isRequired,
      label: string.isRequired,
      sortable: bool.isRequired
    })
  ).isRequired,
  /** define se é expansivel */
  collapse: arrayOf(
    shape({
      key: string.isRequired,
      label: string.isRequired
    })
  ),
  /** dados a serem mostrados */
  // eslint-disable-next-line react/forbid-prop-types
  data: arrayOf(object).isRequired,
  /** total de linhas */
  totalRows: number.isRequired,
  /** numero de páginas (quando há paginação) */
  numberPages: number.isRequired,
  /** página atual (quando há paginação) */
  currentPage: number,
  /** quantos items mostrado por página */
  perPage: number,
  /** função chamada ao mudar de página */
  onChangePage: func,
  /** função chamada ao ordenar */
  onSort: func,
  className: string,
  /** texto mostrado na opção de mostrar mais itens */
  showMoreItemsText: string,
  /** define se está carregando */
  isLoading: bool,
  /** define se a página está carregando */
  isPageLoading: bool,
  /** mensagem quando a tabela estiver vazia */
  messageWhenEmpty: string
}

DataTable.defaultProps = {
  className: '',
  collapse: [],
  currentPage: 1,
  perPage: 10,
  onChangePage: () => {},
  onSort: () => {},
  showMoreItemsText: 'Mostrar mais',
  isLoading: false,
  isPageLoading: false,
  messageWhenEmpty: null
}
