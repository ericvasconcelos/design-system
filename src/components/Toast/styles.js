import { Icon } from 'components/Icon'
import styled, { css } from 'styled-components'
import { getColorFromName } from 'utils/getColorFromName'
import { getTypography } from 'utils/getTypography'

const getBackgroundColor = ({ styleType, theme }) => {
  const colors = {
    error: getColorFromName({ theme, styleColor: 'error100' }),
    success: getColorFromName({ theme, styleColor: 'success400' }),
    warning: getColorFromName({ theme, styleColor: 'warning200' }),
    info: getColorFromName({ theme, styleColor: 'darkBlueCash200' })
  }

  return colors[styleType]
}

const getPosition = ({ position }) => {
  if (position === 'top-right')
    return css`
      right: 1rem;
    `
  if (position === 'top-left')
    return css`
      left: 1rem;
    `
  return css`
    left: 50%;
    transform: translateX(-50%);
  `
}

export const Wrapper = styled.div`
  ${({ theme }) => getTypography({ theme, styleType: 'body2', styleColor: 'greyCash100' })}
  position: fixed;
  top: ${({ active }) => (active ? '1rem' : '-100vh')};
  ${getPosition}
  z-index: 10;
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.gutters.sm};
  padding: ${({ theme }) => `${theme.padding.md} ${theme.padding.lg}`};
  background-color: ${({ theme, styleType }) => getBackgroundColor({ theme, styleType })};
  border-radius: 8px;
  transition: top 0.5s;
  max-width: 450px;
  user-select: none;
`

export const IconStyled = styled(Icon)`
  min-width: 24px;
`

export const CloseButton = styled.button`
  cursor: pointer;
  margin: auto 0;
  height: 16px;
`
