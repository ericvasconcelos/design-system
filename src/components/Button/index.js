import React, { forwardRef } from 'react'

import dynamic from 'next/dynamic'
import { bool, func, node, oneOf, string, number } from 'prop-types'
import { buttonStyles } from 'utils/getButtonStyle/data'
import { getIconNames } from 'utils/getIconNames'

import { Wrapper } from './styles'

const Icon = dynamic(() => import('../Icon/index.js').then(module => module.Icon), {
  ssr: false
})

/**
 * O botão é o componente usado para dar ação para alguma coisa
 */

export const Button = forwardRef(
  (
    {
      borderRadius,
      styleType,
      styleColor,
      children,
      disabled,
      id,
      icon,
      iconSize,
      isLoading,
      onClick,
      onBlur,
      size,
      type,
      expanded,
      ariaLabel,
      loadingText,
      className,
      tabIndex
    },
    ref
  ) => (
    <Wrapper
      ref={ref}
      type={type}
      borderRadius={borderRadius}
      styleColor={styleColor}
      data-testid="Button"
      disabled={disabled || isLoading}
      onClick={onClick}
      styleType={styleType}
      size={size}
      isLoading={isLoading}
      expanded={expanded}
      id={id}
      tabIndex={tabIndex}
      className={className}
      {...(ariaLabel && { 'aria-label': ariaLabel })}
      {...(onBlur && { onBlur })}
    >
      {children && !isLoading ? children : false}
      {loadingText && isLoading ? loadingText : null}
      {(icon || isLoading) && !loadingText && (
        <Icon
          icon={isLoading ? 'loader' : icon}
          styleColor={buttonStyles[styleType][styleColor].fontColor}
          size={iconSize}
        />
      )}
    </Wrapper>
  )
)

Button.propTypes = {
  /**
   * Conteúdo textual do componente
   */
  children: node,
  /**
   * Cor dos textos utilizados no componente
   */
  styleColor: oneOf(['blue', 'white', 'dark']),
  /**
   * Arredondamento do componente
   */
  borderRadius: oneOf(['default', 'rounded']),
  /**
   * Ícone recebido pelo componente
   */
  icon: oneOf([...getIconNames(), '']),
  /**
   * Tamanho do ícone
   */
  iconSize: oneOf([12, 16, 24, 32, 48, 64]),
  /**
   * Estado de 'carregando' do componente
   */
  isLoading: bool,
  /**
   * Estado de habilitado, caso desabilitado ele perde a função de click e
   * enter quando estiver focado
   */
  disabled: bool,
  /**
   * Função que será executada quando o componente for clicado
   */
  onClick: func.isRequired,
  /**
   * Função que será executada quando o componente perder o foco
   */
  onBlur: func,
  /**
   * Tipo do componente
   */
  styleType: oneOf(['primary', 'secondary', 'link', 'text']),
  /**
   * Tamanho do componente
   */
  size: oneOf(['small', 'medium', 'large']),
  /**
   * ID para o componente
   */
  id: string,
  /**
   * Define se o botão vai utilizar todo o espaço do pai ou se adaptará com o
   * conteúdo utilizado como children
   */
  expanded: bool,
  /**
   * Tipo de componente
   */
  type: oneOf(['button', 'submit']),
  /**
   * Texto recebido quando o componente estiver em estado de loading e NÃO receber nenhum icon
   */
  loadingText: string,
  /**
   * Opção para o componente ser utilizado dentro de um styled component
   */
  className: string,
  /**
   * Aria label do componente
   */
  ariaLabel: string,
  /**
   * TabIndex do componente
   */
  tabIndex: number
}

Button.defaultProps = {
  children: '',
  borderRadius: 'default',
  styleColor: 'blue',
  disabled: false,
  isLoading: false,
  icon: 'arrowRight',
  iconSize: 24,
  size: 'small',
  styleType: 'primary',
  id: '',
  expanded: true,
  type: 'button',
  loadingText: '',
  className: '',
  ariaLabel: '',
  onBlur: null,
  tabIndex: 0
}
