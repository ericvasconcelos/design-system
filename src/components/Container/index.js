import React from 'react'

import { node, string } from 'prop-types'

import { Wrapper } from './styles'

export const Container = ({ children, className }) => (
  <Wrapper data-testid="Container" className={className}>
    {children}
  </Wrapper>
)

Container.propTypes = {
  children: node.isRequired,
  className: string
}

Container.defaultProps = {
  className: ''
}
