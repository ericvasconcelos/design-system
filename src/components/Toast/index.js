import React, { useEffect } from 'react'

import { Text } from 'components/Text'
import { oneOf, string, bool, number, func, node } from 'prop-types'
import { theme } from 'styles/theme'

import { Wrapper, IconStyled, CloseButton } from './styles'
import { ToastContextProvider } from './ToastProvider'

export const Toast = ({
  active,
  position,
  styleType,
  text,
  seconds,
  onTimeout,
  icon,
  iconStyleColor,
  isClosable,
  className,
  children
}) => {
  const getStyleColor = () => {
    const colors = {
      error: 'error100',
      success: 'success400',
      warning: 'warning200',
      info: 'darkBlueCash200'
    }

    return colors[styleType]
  }

  useEffect(() => {
    let timeout
    if (active && seconds) {
      timeout = setTimeout(() => {
        onTimeout()
      }, seconds * 1000)
    }

    return () => {
      clearTimeout(timeout)
    }
  }, [active, seconds])

  return (
    <ToastContextProvider>
      <>
        <Wrapper
          data-testid="Toast"
          className={className}
          styleType={styleType}
          active={active}
          position={position}
        >
          {icon && (
            <IconStyled
              icon={icon}
              styleColor={iconStyleColor || getStyleColor({ theme, styleColor: styleType })}
            />
          )}
          <Text styleType="body2" styleColor="greyCash100">
            {text}
          </Text>
          {isClosable && (
            <CloseButton onClick={clearTimeout(seconds)}>
              <img src={theme.icons.close} alt="" />
            </CloseButton>
          )}
        </Wrapper>
        {children}
      </>
    </ToastContextProvider>
  )
}

Toast.propTypes = {
  active: bool,
  position: oneOf(['top-left', 'top-right', 'top-center']),
  styleType: oneOf(['success', 'error', 'warning', 'info']).isRequired,
  text: string.isRequired,
  seconds: number,
  onTimeout: func,
  icon: string,
  iconStyleColor: string,
  isClosable: bool,
  children: node,
  className: string
}

Toast.defaultProps = {
  active: false,
  position: 'top-right',
  seconds: null,
  onTimeout: null,
  icon: '',
  iconStyleColor: null,
  className: '',
  children: null,
  isClosable: false
}
