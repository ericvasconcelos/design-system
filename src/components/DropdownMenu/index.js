import React, { useState } from 'react'

import { Button } from 'components/Button'
import { Popper } from 'components/Popper'
import PropTypes, { arrayOf, bool, string, func } from 'prop-types'

import { Wrapper } from './styles'
/**
 * DropDownMenu é um componente de
 * menu dropdown controlado por um botão.
 * O componente renderiza uma lista de elementos
 * passados por quen estancia o componente.
 */
export const DropdownMenu = ({
  id,
  buttonLabel,
  buttonIcon,
  buttonStyle,
  className,
  expanded,
  menuItems
}) => {
  const [show, setShow] = useState(false)

  const toggleVisibility = () => {
    setShow(!show)
  }

  return (
    <Popper showArrow show={show} arrowBorderColor="greyCash300">
      {reference => (
        <Button
          ref={referenceNode => {
            reference(referenceNode)
          }}
          expanded={expanded}
          icon={buttonIcon}
          styleType={buttonStyle}
          onClick={toggleVisibility}
          aria-haspopup
          className={className}
          aria-expanded={show}
          aria-describedby="initInstr"
          aria-owns={`result${id}`}
          aria-autocomplete="both"
          aria-activedescendant=""
          id={id}
          tabIndex={-1}
        >
          {buttonLabel}
        </Button>
      )}
      <Wrapper id={`result${id}`} aria-labelledby={id} role="listbox">
        {menuItems && menuItems.map(item => <li key={item.key}>{item.render()}</li>)}
      </Wrapper>
    </Popper>
  )
}

DropdownMenu.propTypes = {
  /**
   * buttonLabel é o texto que aparece
   * no botão que controla o menu
   */
  buttonLabel: string,
  /**
   * buttonIcon é o ícone que aparece
   * no botão que controla o menu
   */
  buttonIcon: string,
  /**
   * buttonStyle é o styleType
   * do botão que controla o menu
   */
  buttonStyle: string,
  /**
   * className se refere a prop className
   * do botão que controla o menu
   */
  className: string,
  /**
   * expanded se refere a prop expanded
   * do botão que controla o menu
   */
  expanded: bool,
  /**
   * id será passado ao
   * do botão que controla o menu
   */
  id: string.isRequired,
  /**
   * é uma lista de nodes
   * que farão parte do menu
   */
  menuItems: arrayOf(
    PropTypes.shape({
      key: string,
      render: func
    })
  )
}

DropdownMenu.defaultProps = {
  buttonLabel: '',
  buttonIcon: '',
  buttonStyle: 'primary',
  className: '',
  expanded: false,
  menuItems: []
}
