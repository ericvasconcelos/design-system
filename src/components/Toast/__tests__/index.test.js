import React from 'react'

import { waitFor } from '@testing-library/dom'
import { Toast } from 'components/Toast'
import { styledRender } from 'utils/styledRender'

jest.useFakeTimers()

const TestComponent = () => <h2>i am just a test component</h2>

describe('Toast Component', () => {
  const onTimeout = jest.fn()

  it('component styleType success and inactive', async () => {
    const { container, getByTestId } = styledRender(
      <Toast active={false} styleType="success" text="Message" seconds={2} onTimeout={onTimeout}>
        <TestComponent />
      </Toast>
    )

    await waitFor(() => expect(getByTestId('Toast')).toHaveStyleRule('top', '-100vh'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('component styleType success and active is render', async () => {
    const { container, getByTestId } = styledRender(
      <Toast active styleType="success" text="Message" seconds={2} onTimeout={onTimeout}>
        <TestComponent />
      </Toast>
    )

    await waitFor(() => expect(getByTestId('Toast')).toHaveStyleRule('top', '1rem'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('component styleType error and active is render', async () => {
    const { container, getByTestId } = styledRender(
      <Toast active styleType="error" text="Message" seconds={2} onTimeout={onTimeout}>
        <TestComponent />
      </Toast>
    )

    await waitFor(() => expect(getByTestId('Toast')).toHaveStyleRule('top', '1rem'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('component styleType info and active is render', async () => {
    const { container, getByTestId } = styledRender(
      <Toast active styleType="info" text="Message" seconds={2} onTimeout={onTimeout}>
        <TestComponent />
      </Toast>
    )

    await waitFor(() => expect(getByTestId('Toast')).toHaveStyleRule('top', '1rem'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('component position top left is render', async () => {
    const { container, getByTestId } = styledRender(
      <Toast
        active
        styleType="info"
        position="top-left"
        text="Message"
        seconds={2}
        onTimeout={onTimeout}
      >
        <TestComponent />
      </Toast>
    )

    await waitFor(() => expect(getByTestId('Toast')).toHaveStyleRule('left', '1rem'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('component position top center is render', async () => {
    const { container, getByTestId } = styledRender(
      <Toast
        active
        styleType="info"
        position="top-center"
        text="Message"
        seconds={2}
        onTimeout={onTimeout}
      >
        <TestComponent />
      </Toast>
    )

    await waitFor(() => expect(getByTestId('Toast')).toHaveStyleRule('left', '50%'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('component call callback on timeout', () => {
    styledRender(
      <Toast active styleType="success" text="Message" seconds={2} onTimeout={onTimeout}>
        <TestComponent />
      </Toast>
    )

    jest.runAllTimers()

    expect(onTimeout).toBeCalledWith()
  })
})
