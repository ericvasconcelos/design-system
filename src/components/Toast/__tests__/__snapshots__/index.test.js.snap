// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component component position top center is render 1`] = `
.c1 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  position: fixed;
  top: 1rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem 1.5rem;
  background-color: #455A73;
  border-radius: 8px;
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
  max-width: 450px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  data-testid="Toast"
>
  
  <p
    class="c1"
    data-testid="Text"
  >
    Message
  </p>
</div>
`;

exports[`Toast Component component position top left is render 1`] = `
.c1 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  position: fixed;
  top: 1rem;
  left: 1rem;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem 1.5rem;
  background-color: #455A73;
  border-radius: 8px;
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
  max-width: 450px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  data-testid="Toast"
>
  
  <p
    class="c1"
    data-testid="Text"
  >
    Message
  </p>
</div>
`;

exports[`Toast Component component styleType error and active is render 1`] = `
.c1 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  position: fixed;
  top: 1rem;
  right: 1rem;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem 1.5rem;
  background-color: #FFCECE;
  border-radius: 8px;
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
  max-width: 450px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  data-testid="Toast"
>
  
  <p
    class="c1"
    data-testid="Text"
  >
    Message
  </p>
</div>
`;

exports[`Toast Component component styleType info and active is render 1`] = `
.c1 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  position: fixed;
  top: 1rem;
  right: 1rem;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem 1.5rem;
  background-color: #455A73;
  border-radius: 8px;
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
  max-width: 450px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  data-testid="Toast"
>
  
  <p
    class="c1"
    data-testid="Text"
  >
    Message
  </p>
</div>
`;

exports[`Toast Component component styleType success and active is render 1`] = `
.c1 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  position: fixed;
  top: 1rem;
  right: 1rem;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem 1.5rem;
  background-color: #01B36A;
  border-radius: 8px;
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
  max-width: 450px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  data-testid="Toast"
>
  
  <p
    class="c1"
    data-testid="Text"
  >
    Message
  </p>
</div>
`;

exports[`Toast Component component styleType success and inactive 1`] = `
.c1 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0 {
  color: #FFFFFF;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  position: fixed;
  top: -100vh;
  right: 1rem;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  padding: 1rem 1.5rem;
  background-color: #01B36A;
  border-radius: 8px;
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
  max-width: 450px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  data-testid="Toast"
>
  
  <p
    class="c1"
    data-testid="Text"
  >
    Message
  </p>
</div>
`;
